src/browser/components/AddressBar/AddressBar.js

```diff
@@ -26,6 +28,16 @@ export default class AddressBar extends React.Component {
       this.input.value = this.props.query;
       this.input.blur();
     }
+
+    if (this.props.tabCount > prevProps.tabCount) {
+      // Focus the address field after opening a new tab
+      this.input.select();
+    } else if (this.props.tabCount < prevProps.tabCount) {
+      // Blur the address field after closing a tab and make sure the
+      // correspondent address is shown
+      this.input.value = this.props.query;
+      this.input.blur();
+    }
```

Comment:
I'm sorry for the pushback on this.  I've been thinking about this a bit since we picked our design competition winner.  With the new design, we're going to be inverting the tabs and address bar, such that the tabs appear on top, and the address bar on bottom (similar to Chrome).  What this means is that we would probably end up rendering one `AddressBar` component per `DAppContainer` (similar to the `webview` element & new `Error` component you added).  When this happens, I think it will simplify things quite a bit, such that we can simply use `componentDidMount` for this behavior.

Given that, I'd rather punt a little bit on this, until after we flip the tabs & address bar around.